### PR TITLE
More Testcases

### PR DESCRIPTION
**Pull request by ruihan00**

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonAdaptedTag.java [hunk 1]
```diff
@@ -1,51 +1,51 @@
...
-import com.fasterxml.jackson.annotation.JsonValue;
-
-import seedu.address.commons.exceptions.IllegalValueException;
-import seedu.address.model.entry.EntryType;
-import seedu.address.model.tag.Tag;
-
-/**
- * Jackson-friendly version of {@link Tag}.
- */
-class JsonAdaptedTag {
```
Comment (ruihan00) [inline src/main/java/seedu/address/storage/JsonAdaptedTag.java:13]: I think we don't need this file anymore

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -231,7 +231,7 @@ public ObservableList<PieChart.Data> getIncomePieChartData() {
             new PieChart.Data("Gifts", incomePieChartArr[4]), new PieChart.Data("Others", incomePieChartArr[5]));
 
 
-        return incomePieChartData;
+        return FXCollections.unmodifiableObservableList(incomePieChartData);
```
Comment (ruihan00) [inline src/main/java/seedu/address/model/ModelManager.java:234]: made the ObservableList here unmodifiable similar to the expenditure and income list

File: src/main/java/seedu/address/model/ModelManager.java [hunk 3]
```diff
@@ -275,7 +275,7 @@ public ObservableList<PieChart.Data> getExpensePieChartData() {
             new PieChart.Data("Education", expensePieChartArr[3]), new PieChart.Data("Housing", expensePieChartArr[4]),
             new PieChart.Data("Others", expensePieChartArr[5]));
 
-        return expensePieChartData;
+        return FXCollections.unmodifiableObservableList(expensePieChartData);
```
Comment (ruihan00) [inline src/main/java/seedu/address/model/ModelManager.java:278]: same as prev comment

File: src/test/data/JsonSerializablePennyWiseTest/typicalPersonsAddressBook.json [hunk 4]
```diff
@@ -16,5 +16,15 @@
     "date" : "21-08-2022",
     "tagged" : "Entertainment"
   }],
-  "incomes" : []
+  "incomes" : [{
```
Comment (ruihan00) [inline src/test/data/JsonSerializablePennyWiseTest/typicalPersonsAddressBook.json:19]: will do a refractor of the files here to remove persons in the next pr

**Review by JasonYapzx — APPROVED** [review]
LGTM, very exhaustive set of test cases,

**Pull request merged**